src/main/scala/devices/debug/Debug.scala

```diff
@@ -722,6 +721,263 @@ class TLDebugModuleInner(device: Device, getNComponents: () => Int, beatBytes: I
       }
     }
 
+    //--------------------------------------------------------------
+    // System Bus Access
+    //--------------------------------------------------------------
+    // --- SBCS Registers ---
+
+    sb2tlOpt match {                   
+      case Some(sb2tl) => {
+       
+        /* Flags for setting error flag registers, needed because the number of SBDATA and
+         * SBADDRESS registers depends on system bus widths
+         */
+        val anyAddressWrEn = Wire(init = false.B)
+        val anyDataRdEn    = Wire(init = false.B)
+        val anyDataWrEn    = Wire(init = false.B)
+
+        // --- SBCS Status Register ---
+        val SBCSFieldsReg      = Reg(new SBCSFields())
+        val SBCSFieldsRegReset = Wire(init = (new SBCSFields()).fromBits(0.U))
+        val SBCSRdData         = Wire(init = new SBCSFields().fromBits(0.U)) 
+        val SBCSWrDataVal      = Wire(init = 0.U(32.W))
+        val SBCSWrData         = Wire(init = new SBCSFields().fromBits(SBCSWrDataVal))
+        val SBCSRdEn, SBCSWrEn = Wire(init = false.B)
+
+
+        // --- System Bus Registers ---
+        val SBADDRESS0FieldsReg = Reg(new SBADDRESS0Fields())
+        val SBADDRESS1FieldsReg = Reg(new SBADDRESS1Fields())
+        val SBADDRESS2FieldsReg = Reg(new SBADDRESS2Fields())
+        val SBADDRESS3FieldsReg = Reg(new SBADDRESS3Fields())
+        val SBADDRESS0RdData    = Wire(init = new SBADDRESS0Fields().fromBits(0.U))
+        val SBADDRESS1RdData    = Wire(init = new SBADDRESS1Fields().fromBits(0.U))
+        val SBADDRESS2RdData    = Wire(init = new SBADDRESS2Fields().fromBits(0.U))
+        val SBADDRESS3RdData    = Wire(init = new SBADDRESS3Fields().fromBits(0.U))
+
+        val SBADDRESS0WrData, SBADDRESS1WrData, SBADDRESS2WrData, SBADDRESS3WrData = Wire(init = 0.U(32.W))
+        val SBADDRESS0RdEn,   SBADDRESS1RdEn,   SBADDRESS2RdEn,   SBADDRESS3RdEn   = Wire(init = false.B)  
+        val SBADDRESS0WrEn,   SBADDRESS1WrEn,   SBADDRESS2WrEn,   SBADDRESS3WrEn   = Wire(init = false.B)  
+
+        val autoIncrementedAddr = Wire(init = 0.U(128.W))
+        autoIncrementedAddr := sb2tl.module.io.addrIn + (1.U << SBCSFieldsReg.sbaccess)
+
+        // ADDR0 Register is required
+        when (~io.dmactive) {
```

Comment:
RE my above comment, then this code would become something like 

```
SBADDRESSRegs.zipWithIndex.foreach { case (a, i) => 
  when (~io.dmactive) {
     a := 0.U
  }.otherwise {
     a:= Mux(SBADDRESSWrEn(i) && ...)
 }

...
}
```